### DISCUSSION TITLE
Expose Type Aliases

### DISCUSSION THREAD

**Discussion by JEdmario16** [spyoungtech/ahk#287] [Q&A] [answered]
Hello everyone. In my application, I have the following setup:

```python
class MacroController:
    def __init__(self):
       self.ahk = AHK()
       targets =  ["ToolTip", "Pixel", "Mouse", "Caret", "Menu"]
       for target in targets:
           self.ahk.set_coord_mode(target=target, relative_to="Window")
```
The problem arises because the `target` iterator variable will be of type str, and mypy will throw errors at me because it must be one of:
```python
Union[
    Literal["ToolTip"],
    Literal["Pixel"],
    Literal["Mouse"],
    Literal["Caret"],
    Literal["Menu"],
]
```
To resolve this, I need to ensure that my targets list has the same definition(or define the same type alias by my own). It would look something like this:
```python
targets: list[
    Union[
        Literal["ToolTip"],
        Literal["Pixel"],
        Literal["Mouse"],
        Literal["Caret"],
        Literal["Menu"],
    ]
] = ["ToolTip", "Pixel", "Mouse", "Caret", "Menu"]
```

It would be even better if I could import the CoordModeTargets type alias from _engine.py, but unfortunately, it's not exposed and i cant import by default.

**Comment by JEdmario16** [comment, marked as answer]
Update: I just discovered that i can import tipes alises(or anything that i want wich is in engine.py) by using `from ahk._sync.engine import CoordModeTargets` or anything else

**Reply by spyoungtech** [reply]
Thanks for the question. This is something I've been looking at improving so you don't need to reach into the internal modules for this stuff. In a very recent release, we did this for the various exception classes. I'm planning to expose types in a more convenient way in a future release, too. Importing the types as you have will continue to be supported, as well.

**Reply by spyoungtech** [reply]
In 1.7.0 most type aliases are now exposed at the package top level.
